crates/ide_completion/src/completions/record.rs

```diff
@@ -62,14 +81,26 @@ pub(crate) fn complete_record_literal(
         return None;
     }
 
-    if let hir::Adt::Struct(strukt) = ctx.expected_type.as_ref()?.as_adt()? {
-        if ctx.path_qual().is_none() {
-            let module = if let Some(module) = ctx.module { module } else { strukt.module(ctx.db) };
-            let path = module.find_use_path(ctx.db, hir::ModuleDef::from(strukt));
+    match ctx.expected_type.as_ref()?.as_adt()? {
+        hir::Adt::Struct(strukt) => {
+            if ctx.path_qual().is_none() {
```

Comment:
```suggestion
        hir::Adt::Struct(strukt) if ctx.path_qual().is_none() => {
```
can be moved into the guard, likewise for the union arm